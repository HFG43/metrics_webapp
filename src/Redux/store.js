import { configureStore } from '@reduxjs/toolkit';
import { logger } from 'redux-logger';
import marketReducer from './MarketSlice/MarketSlice';
import companiesReducer from './CompaniesSlice/CompaniesSlice';

const store = configureStore({
  reducer: {
    market: marketReducer,
    companies: companiesReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
