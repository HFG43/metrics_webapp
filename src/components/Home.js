import CompaniesListDisplay from './CompaniesListDisplay';
import MarketView from './MarketView';

const Home = () => (
  <>
    <MarketView />
    <CompaniesListDisplay />
  </>
);
export default Home;
