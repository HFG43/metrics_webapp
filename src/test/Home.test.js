import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from './StoreMock';
import Home from '../components/Home';

describe('Home', () => {
  test('Home page renders correctly', () => {
    const tree = render(
      <MemoryRouter initialEntries={['/home']}>
        <Provider store={store}>
          <Home />
        </Provider>
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
