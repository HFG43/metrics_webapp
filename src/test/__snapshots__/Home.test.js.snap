// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        Retrieving S&P 500 price... Please wait
      </div>
      <h3
        class="company_price_header"
      >
        PRICE BY COMPANY
      </h3>
      <div
        class="form"
      >
        <input
          id="search"
          pattern="^[A-Za-z0-9]+([ ]?[A-Za-z0-9]+)?$"
          placeholder="Search for your company"
          type="text"
        />
      </div>
      <div
        class="companies_list_container"
      >
        <a
          href="/company/Test"
        >
          <article
            class="obsc"
          >
            <div>
              <div
                class="name_arrow_container"
              >
                <img
                  alt="Arrow icon"
                  class="arrow"
                  src="arrow_next.svg"
                />
                <div
                  class="name_container"
                >
                  <span
                    class="company_name"
                  >
                    Company Test
                  </span>
                </div>
              </div>
              <div
                class="symbol_price_container"
              >
                <p>
                  Test
                </p>
                <span>
                  181.99
                </span>
              </div>
            </div>
          </article>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      Retrieving S&P 500 price... Please wait
    </div>
    <h3
      class="company_price_header"
    >
      PRICE BY COMPANY
    </h3>
    <div
      class="form"
    >
      <input
        id="search"
        pattern="^[A-Za-z0-9]+([ ]?[A-Za-z0-9]+)?$"
        placeholder="Search for your company"
        type="text"
      />
    </div>
    <div
      class="companies_list_container"
    >
      <a
        href="/company/Test"
      >
        <article
          class="obsc"
        >
          <div>
            <div
              class="name_arrow_container"
            >
              <img
                alt="Arrow icon"
                class="arrow"
                src="arrow_next.svg"
              />
              <div
                class="name_container"
              >
                <span
                  class="company_name"
                >
                  Company Test
                </span>
              </div>
            </div>
            <div
              class="symbol_price_container"
            >
              <p>
                Test
              </p>
              <span>
                181.99
              </span>
            </div>
          </div>
        </article>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
